### PR TITLE
Mhv 59277 contact list scaffolding

### PR DESCRIPTION
**Pull request by KyleCardwell**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

- Started scaffolding for the new Contact list page for Secure Messaging.  The page is hidden behind a feature flag (mhvSecureMessagingEditContactList), and is not connected to the API yet.  Currently, it will load the triage teams for a user and organize them by facility.  Each facility has a "Select all teams" checkbox, that selects only the teams from that facility.

### - [MHV-59277](https://jira.devops.va.gov/browse/MHV-59277) - Create scaffolding for new page for Contact list ###

> Create scaffolding for new page for Contact list (Add/edit)
> 
> Question:  Does this page require a new URL?  From Emily on 7/22, Yes this will require a new URL.
> 
> 1- Create new page for Contact list (Note: There are 2 different flows in FIGMA)
> 
> Users with more that one system
> Users with only one system
> 
> Complete/latest requirements located in Figma:  https://www.figma.com/design/uddjxccqvUyz5FSg1aD5wi/ZFeat-{}Triage-Group-Selection-Flows?node-id=792-21547&t=1UMB6N8tImODPlq8{-}
> 
> User Story: As a Secure Messaging User, I want to be able to add or edit my contact list in the instance when I do not see the group I am looking for.
> 
> Given: The user is unable to locate the needed group in the dropdown list in the "To" field
> When: The user can click on the link, "If you can't find your team" and get more information and help to navigate the problem
> Then:  The user can navigate to the Contact list page and add/edit the groups the user is associate with, save changes and see a success banner. 
> 
> Feature Flag Y/N ? Y-For all contact list functionality
> Manual Testing Y/N ? Y
> Automated Testing Y/N ? Y
> Accessibility Testing Y/N ? Y
> 
> UCD Validation Y/N ? Y
> 
> Acceptance Criteria:
> AC1 New page created for Contact list which will accommodate both flows
> AC2  New URL for this page
> AC3  UCD validation is complete
> AC4 
> 
> Links to UCD: 
> FIGMA Link: https://www.figma.com/design/uddjxccqvUyz5FSg1aD5wi/ZFeat---Triage-Group-Selection-Flows?node-id=792-21547&t=1UMB6N8tImODPlq8-4

## Testing done

- Manual testing and started unit tests (more to come when more functionality is added)

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |Screenshot 2024-07-31 at 9 39 07 AM|
| Desktop |        |Screenshot 2024-07-31 at 9 38 41 AM|
| Desktop |        |Screenshot 2024-07-31 at 9 50 37 AM|
| Desktop |        |Screenshot 2024-07-31 at 9 51 39 AM|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

AC1 New page created for Contact list which will accommodate both flows
AC2  New URL for this page
AC3  UCD validation is complete 

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/containers/AuthorizedRoutes.jsx [hunk 1]
```diff
@@ -74,6 +79,11 @@ const AuthorizedRoutes = () => {
         >
           <FolderThreadListView />
         </AppRoute>
+        {contactListPage && (
+          <AppRoute exact path={Paths.CONTACT_LIST} key="ContactList">
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/containers/AuthorizedRoutes.jsx:83]: this is not impactful, but for consistency, we may want to keep `key` values the same as of the child component i.e. `EditContactList`. But again, it is not criticial and does not impact routing in any way

File: src/applications/mhv-secure-messaging/components/FacilityCheckboxGroup.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <div className="vads-u-margin-left--2 small-screen:vads-u-margin-left--3">
+          {triageTeams.map(team => {
+            return (
+              <VaCheckbox
+                data-testid="contact-list-select-team"
+                key={team.triageTeamId}
+                label={team.name}
+                message-aria-describedby={`Select ${team.name}`}
+                checked={team.preferredTeam}
+                onVaChange={() => updatePreferredTeam(team.triageTeamId)}
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/FacilityCheckboxGroup.jsx:58]: it looks like the `updatePreferredTeam` function should accept 2 arguments

**Review by alex-morgun-1 — COMMENTED** [review]
Great work. I left couple comments

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
Looks good. I left couple comments

File: src/applications/mhv-secure-messaging/containers/AuthorizedRoutes.jsx [hunk 3]
```diff
@@ -28,6 +30,9 @@ AppRoute.propTypes = {
 };
 
 const AuthorizedRoutes = () => {
+  const contactListPage = useSelector(
+    state => state.featureToggles.mhvSecureMessagingEditContactList,
+  );
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/containers/AuthorizedRoutes.jsx:35]: I think you may want to use this approach to look up the feature toggle value
```suggestion
 const contactListPage = useSelector(
    state =>
      state.featureToggles[
        FEATURE_FLAG_NAMES.mhvSecureMessagingEditContactList
      ],
  );

```

File: src/applications/mhv-secure-messaging/components/FacilityCheckboxGroup.jsx (hunk 2 above)
Comment (KyleCardwell) [reply src/applications/mhv-secure-messaging/components/FacilityCheckboxGroup.jsx:58]: Good catch!  I'll update that.

**Review by vsaleem — APPROVED** [review]

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**